content/blog/create-a-general-purpose-kubeconfig-file-in-hpe-greenlake-for-containers.md

```diff
@@ -0,0 +1,261 @@
+
+
+
+```bash
+$ kubectl describe serviceaccounts cfe-demo-sa 
+Name:                cfe-demo-sa
+Namespace:           cfe-demo-cluster
+Labels:              <none>
+Annotations:         Image pull secrets:  <none>
+Mountable secrets:   cfe-demo-sa-token-2zlzf
+Tokens:              cfe-demo-sa-token-2zlzf
+Events:              <none>
+
+$ kubectl describe secrets cfe-demo-sa-token-2zlzf 
+Name:         cfe-demo-sa-token-2zlzf
+Namespace:    cfe-demo-cluster
+Labels:       <none>
+Annotations:  kubernetes.io/service-account.name: cfe-demo-sa
+              kubernetes.io/service-account.uid: a467f9bd-655d-413f-ad77-5156d03d2322
+
+Type:  kubernetes.io/service-account-token
+
+Data
+====
+ca.crt:     1066 bytes
+namespace:  16 bytes
+token:      iIsImtpZCI6IjA2YnhmSVZrVDRGWnBab0VOYXhnWFBTTE1WWmptUm40eER
+
+```
+
+Note that if you use `-o yaml` instead of `describe` in the commands, you get a base64-encoded version of the token. You must decode it before you use it.
+
+
+
+If you access the Kubernetes API directly, e.g., from `curl`, you can use the token as the bearer token for the authorization header. 
+
+However, if you have your scripts running outside the cluster that use kubectl or a client library to access the Kubernetes cluster, you need the kubeconfig file to load configs from. You follow up the following section to create such a kubeconfig file.
+
+
+
+### Create a Kubeconfig File
+Here is a shall script to create a kubeconfig file using the secret token of the service account. Update those variables in the script header to match with your environment.
+
+
+
+```bash
+# create-kubeconfig.sh
+
+# Update those variables to match your environment
+SERVICE_ACCOUNT_NAME="cfe-demo-sa"
+CONTEXT=$(kubectl config current-context)
+NEW_CONTEXT="cfe-demo-context"
+TOKEN_USER="cfe-token-user"
+KUBECONFIG_FILE="kubeconfig-sa"
+
+
+# Extract service account token
+SECRET_NAME=$(kubectl get serviceaccount ${SERVICE_ACCOUNT_NAME} --context ${CONTEXT}  -o jsonpath='{.secrets[0].name}')
+TOKEN_DATA=$(kubectl get secret ${SECRET_NAME}  --context ${CONTEXT}  -o jsonpath='{.data.token}')
+TOKEN=$(echo ${TOKEN_DATA} | base64 -d)
+
+#Create a general-purpose kubeconfig file
+kubectl config view --raw > tmp.raw
+kubectl --kubeconfig tmp.raw config use-context ${CONTEXT}
+kubectl --kubeconfig tmp.raw config view --flatten --minify > tmp.min
+kubectl config --kubeconfig tmp.min rename-context ${CONTEXT} ${NEW_CONTEXT}
+kubectl config --kubeconfig tmp.min  set-credentials ${TOKEN_USER} --token ${TOKEN}
+kubectl config --kubeconfig tmp.min set-context ${NEW_CONTEXT} --user ${ TOKEN_USER }
+kubectl config --kubeconfig tmp.min view --flatten --minify > ${KUBECONFIG_FILE}
+
+# Cleanup tmp
+rm tmp.raw
+rm tmp.min
+
+```
+After running the script, a general-purpose kubeconfig file `kubeconfig-sa` is created. Export the kubeconfig file as the environment variable `KUBECONFIG`, you can access the Kubernetes cluster and check all the resources deployed in the cluster.
```

Comment:
Change the beginning of the second sentence to read:
"By exporting the kubeconfig file as..."